pkg/cluster/majorversionupgrade.go

```diff
@@ -106,13 +107,35 @@ func (c *Cluster) majorVersionUpgrade() error {
 			c.eventRecorder.Eventf(c.GetReference(), v1.EventTypeNormal, "Major Version Upgrade", "Starting major version upgrade on pod %s of %d pods", masterPod.Name, numberOfPods)
 			upgradeCommand := fmt.Sprintf("/usr/bin/python3 /scripts/inplace_upgrade.py %d 2>&1 | tee last_upgrade.log", numberOfPods)
 
-			result, err := c.ExecCommand(podName, "/bin/su", "postgres", "-c", upgradeCommand)
-			if err != nil {
-				c.eventRecorder.Eventf(c.GetReference(), v1.EventTypeNormal, "Major Version Upgrade", "Upgrade from %d to %d FAILED: %v", c.currentMajorVersion, desiredVersion, err)
-				return err
+			// ######## checking if the spilo image runs with root or non-root (check for user id=0)
+			c.logger.Infof("checking if the spilo image runs with root or non-root (check for user id=0)")
+
+			resultIdCheck, errIdCheck := c.ExecCommand(podName, "/bin/bash", "-c", "/usr/bin/id -u")
+			if errIdCheck != nil {
+				c.eventRecorder.Eventf(c.GetReference(), v1.EventTypeNormal, "Major Version Upgrade", "Pre-Upgrade step (checking user with id -u) for Upgrade from %d to %d FAILED: %v", c.currentMajorVersion, desiredVersion, errIdCheck)
+				return errIdCheck
+			}
+
+			resultIdCheck = strings.TrimSuffix(resultIdCheck, "\n")
+
+			if resultIdCheck != "0" {
```

Comment:
on set the command here ("/bin/bash" or "/bin/su") so you don't repeat all the code in this if else block